src/types.tsx

```diff
@@ -713,3 +713,24 @@ export interface SearchBarProps {
    */
   shouldShowHintSearchIcon?: boolean;
 }
+
+/**
+ * Since the search bar is a component that is extended from View on iOS,
+ * we can't use onFocus and onBlur events directly there (as of the event naming conflicts).
+ * To omit any breaking changes, we're handling this type to rename onFocus and onBlur events
+ * to onSearchFocus and onSearchBlur inside the native component of the search bar.
+ */
+export type RenamedSearchBarProps = Rename<
+  SearchBarProps,
+  { onFocus: 'onSearchFocus'; onBlur: 'onSearchBlur' }
+>;
+
+/**
+ * Helper type, used to rename certain keys in the interface, given from object.
+ */
+type Rename<
+  T,
+  R extends {
+    [K in keyof R]: K extends keyof T ? PropertyKey : 'Error: key not in T';
+  }
+> = { [P in keyof T as P extends keyof R ? R[P] : P]: T[P] };
```

Comment:
I appreciate the idea here (and like it btw.), however do we need this? *If w/o this fix setting these options would crash application anyway* I believe we can safely treat this as a required fix and not a breaking change, thus removing the need for such workaround.

Reply:
We mainly need this, because of the Typescript that is shouting about the unknown prop 🙄 we of course wanna to use onSearchFocus and onSearchBlur prop, since we've got them defined with that name in the spec

Reply:
I think I require an assistance of @WoLewicki there, as I was speaking with him about this change 😁 but when I was wondering about it, I was thinking that it would also require us to introduce the breaking change inside the react-navigation, and also there will be time for introducing such breaking changes (very soon 🤭). What do you think?

Reply:
We don't want to change the user's API for sure so the `SearchBarProps` must stay the same but `NativeSearchBar` should take its props from the spec inside `fabric` directory so we are sure we are sending the proper ones to the native side. Or am I missing something?

Reply:
I agree with you Wojtek and I would totally go into using Spec as the type only for codegen components, instead of reusing types from types.tsx, but tbh I'd leave this task for a separate PR. What do you think?

Reply:
But then you wouldn't have to add this `RenamedSearchBarProps` type at all yeah? But if you want, you can apply it in the follow-up PR. I'd also change all props to have prefix of `onSearch...` to keep the convention and not stumble to some similar issue later.

Reply:
So the conclusion would be to:

1. leave types in `types.tsx` intact (so the public API of our lib)
2. Make `NativeSearchBar` use prop types that do come from Fabric spec, not from the `types.tsx`
3. in `SearchBar` component hijack these event-related props: `onBlur` & `onFocus` and pass them to as values to `onSearchBlur` & `onSearchFocus` props on `NativeSearchBar` respectively

This way we avoid the need for this typescript hacks.

Reply:
Hi @WoLewicki @kkafar, I've just added a change that doesn't impact types file and does a little bit tricky stuff inside the SearchBar, but I think this is fine 😄. Could you tell me what do you think about it? https://github.com/software-mansion/react-native-screens/pull/2154/commits/299ce6a18dae0b04f81b9c49ec5a031ef88e193e